### ISSUE TITLE
Only a single class is being detected

### ISSUE THREAD

**Issue by gaklein2007** [matterport/Mask_RCNN#890]
I am currently adapting this code to be able to handle images with 5 classes in them. After training, only a single class is being identified at all. I am not exactly sure why only one class is being identified. If the input_gt_class_ids is a vector of 5 integers representing the classes, only the last one is being identified (ie, if input_gt_class_ids  = [1, 2, 3, 4, 5], only class 1 is being identified). Has anyone come across this problem? It seems to be a pretty fundamental bug in my code, as no matter what I change the outputs are still like it.

**Comment by fastlater** [comment]
did you set the mask properly. Check your load_mask function. Did you create the class_ids array properly?
If yes, use tensorboard to check the training output variables.

**Comment by gaklein2007** [comment]
It has nothing to do with the mask (I'm not even feeding a mask into it). I am only looking using the detection side of it for now. It has to do with my training. The validation loss functions are not updating that much whereas the training data is. Not sure as to why.

**Comment by anguoyang** [comment]
@kleingeo @fastlater  I have met the same problem, have you solved it? thank you.
There is no problem during training, and the mask result is correct(display on the image), however, they all be identified by the first one, for example, the class name is pan1,pan2...pan30, but when I inference on test.jpg(pan5,pan6,pan7 in it for example), but they all be identified with pan1, it is strange that the mask is correct, actually it is perfect when showing on image

**Comment by anguoyang** [comment]
@fastlater I found that the problem maybe due to mis-calculate the class_ids when loading mask, but actually I modified from balloon example, which has only 1 class, even if I added all my classes during initialization, the class_ids are always [0:1]/1, could you please tell me how to modify the load_mask function when I have more than 1 class? the shapes example is not so good as the data was generated, balloon is better but it has only 1 class

**Comment by anguoyang** [comment]
@waleedka

**Comment by ApoorvaSuresh** [comment]
@kleingeo Same problem here, I train for 3 classes, but on inference mode, it detects just one class. Doesn't detect the other class at all! 

How did you solve it?
 Thank you :)
